src/backend/mpi/swdsm.cpp

```diff
@@ -95,22 +93,20 @@ char * barwindowsused;
 sem_t ibsem;
 
 /*Loading and Prefetching*/
-/** @brief Tracking address of pages that should be loaded by loadthread1 */
-unsigned long *loadline;
-/** @brief Tracking cacheindex of pages that should be loaded by loadthread1 */
-unsigned long *loadtag;
-/** @brief Tracking address of pages that should be loaded by loadthread2 */
-unsigned long *prefetchline;
-/** @brief Tracking cacheindex of pages that should be loaded by loadthread2 */
-unsigned long *prefetchtag;
-/** @brief loadthread1 waits on this to start loading remote pages */
-sem_t loadstartsem;
-/** @brief signalhandler waits on this to complete a transfer */
-sem_t loadwaitsem;
-/** @brief loadthread2 waits on this to start loading remote pages */
-sem_t prefetchstartsem;
-/** @brief signalhandler waits on this to complete a transfer */
-sem_t prefetchwaitsem;
+/**
+ * @brief load into cache helper function
+ * @param tag aligned address to load into the cache
+ * @param line cache entry index to use
+ */
+void load_cache_entry(unsigned long tag, unsigned long line);
```

Comment:
I'd rather the parameters where named something better. It seems the names where taken from the existing code? How about something like `address` and `local_index`?

Reply:
I think this one is important to discuss a little bit: should this even be two parameters?
Note that startIndex (which seeds line) is essentially computed from the alignedDistrAddr (which seeds tag).
See definitions /  computation at
https://github.com/davidklaftenegger/argodsm/blob/ebea0f3cd8e4cf5623b7e362101e36e6aa2df0c0/src/backend/mpi/swdsm.cpp#L317-L324
https://github.com/davidklaftenegger/argodsm/blob/ebea0f3cd8e4cf5623b7e362101e36e6aa2df0c0/src/backend/mpi/swdsm.cpp#L274-L277
and usage at
https://github.com/davidklaftenegger/argodsm/blob/ebea0f3cd8e4cf5623b7e362101e36e6aa2df0c0/src/backend/mpi/swdsm.cpp#L415-L418


Maybe we should restructure this and only have a single parameter, which could either be the aligned or unaligned address of the miss. Opinions?
CC @magnusandersnorgren @pekemark @lundgren87 @kostis

Reply:
As we talked in private, maybe such changes should be left for a future PR.

Reply:
This issue is now tracked at #31